tiledb/sm/c_api/tiledb_experimental.h

```diff
@@ -295,6 +295,41 @@ TILEDB_EXPORT int32_t tiledb_query_get_status_details(
     tiledb_query_t* query,
     tiledb_query_status_details_t* status);
 
+/* ********************************* */
+/*              CONTEXT              */
+/* ********************************* */
+
+/**
+ * Creates a TileDB context, which contains the TileDB storage manager
+ * that manages everything in the TileDB library.
+ *
+ * **Examples:**
+ *
+ * Without config (i.e., use default configuration):
+ *
+ * @code{.c}
+ * tiledb_ctx_t* ctx;
+ * tiledb_error_t* error;
+ * tiledb_ctx_alloc_with_error(NULL, &ctx, &error);
+ * @endcode
+ *
+ * With some config:
+ *
+ * @code{.c}
+ * tiledb_ctx_t* ctx;
+ * tiledb_error_t* error;
+ * tiledb_ctx_alloc(config, &ctx, &error);
+ * @endcode
+ *
+ * @param config The configuration parameters (`NULL` means default).
+ * @param ctx The TileDB context to be created.
+ * @param error Error object returned upon error (`NULL` if there is
+ *     no error).
+ * @return `TILEDB_OK` for success and `TILEDB_OOM` or `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_ctx_alloc_with_error(
```

Comment:
Agree on the `_with_error` not being ideal. However I'd like to keep it as `_with_error` and not `_v2` because I have a few other changes I think we want to make around this for a real `v2`. I consider this a not-hack hack to expose out the error so we can catch to facilitate user experience in 2.7 and iterate for a better item in 2.8/2.9 and drop this completely.

Reply:
The real question is not what happens while it's experimental, but what it looks like when it leaves that status. I'm OK with a "real _v2", in which case I might suggest `..._v2_alpha` for the name in the PR.